mtui/actions/downgrade.py

```diff
@@ -0,0 +1,32 @@
+from string import Template
+
+from ..messages import MissingDowngraderError
+from ..utils import DictWithInjections
+
+
+def zypper() -> dict[str, Template]:
+    list_command_template = r"""
+for p in $packages; do \
+zypper -n se -s --match-exact -t package $$p; \
```

Comment:
`$$p` is that some sort of escaping? I suspect a typo.

Reply:
its escape for `$` in `string.Template`  which uses `$keyword` to fill template :D

Reply:
Hm, I see; and I guess the template system is different than in the original code, which explains why it wasn't escaped before. So now it is populated inside https://github.com/openSUSE/mtui/pull/33/files/c2e6656b3bf62a4eff4c79f1b10c04912418eefc#diff-2f7e2205eec4e38ab21036cf0cb2921a7db2c8de6449b1d36ec26a694015a613R240, so this one is OK, but!